core/domain/suggestion_services.py

```diff
@@ -1267,14 +1267,43 @@ def _update_suggestion_counts_in_community_contribution_stats(
         suggestions, amount)
 
 
-def update_suggestion(suggestion_id, change):
-    """Updates the change of a suggestion with the given suggestion_id.
+def update_translation_suggestion(suggestion_id, translation_html):
+    """Updates the translation_html of a suggestion with the given
+    suggestion_id.
 
     Args:
         suggestion_id: str. The id of the suggestion to be updated.
-        change: obj. The new change object.
+        translation_html: str. The new translation_html string.
     """
     suggestion = get_suggestion_by_id(suggestion_id)
-    suggestion.pre_update_validate(change)
-    suggestion.change = change
+
+    if suggestion.suggestion_type != feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT:
+        raise Exception(
+            'Expected suggestion type is %s.' % (
+                feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT))
+
+    suggestion.change.translation_html = html_cleaner.clean(translation_html)
```

Comment:
Here and below: you can still call pre_update_validate() on the change object, right? (Does it make sense to do that?)

Reply:
I added pre_update_validate() for translations. But I think it doesn't make sense of adding this for questions - as question_dict might not be changed in the update of the question so it throws an error, saying that question_dict is not changed if we call pre_update_validate()

Reply:
Not sure I understand ... why can't you have pre_update_validate() for questions handle that case correctly as well?

Reply:
![Screen Shot 2021-05-15 at 10 01 48 PM](https://user-images.githubusercontent.com/41409951/118371230-3c521b80-b5c9-11eb-8dca-7b802f3e748d.png)
These are the checks made in pre_update_validate() for a question suggestion. Here, question dict is checked whether it is not changed during the update. We have a usecase in this case where the skill difficulty is changed while question dict is unchanged. In that scenario, pre_update_validate() does't allow to update the suggestion as it throws an error because question dict is unchanged.

Reply:
So why not update pre_update_validate() to handle this case correctly? You should probably change the code so that it does the right thing.

Reply:
Just noting that this is the only other thing that needs to be resolved besides the comment above.

Reply:
Done